streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -210,38 +210,39 @@ private RuntimeException closeNonRunningTasks(final Collection<Task> tasks) {
     }
 
     private RuntimeException suspendTasks(final Collection<Task> tasks) {
-        RuntimeException exception = null;
+        final AtomicReference<RuntimeException> exception = new AtomicReference<>(null);
         for (Iterator<Task> it = tasks.iterator(); it.hasNext(); ) {
             final Task task = it.next();
             try {
                 task.suspend();
                 suspended.put(task.id(), task);
             } catch (final TaskMigratedException closeAsZombieAndSwallow) {
                 // as we suspend a task, we are either shutting down or rebalancing, thus, we swallow and move on
-                closeZombieTask(task);
+                exception.compareAndSet(null, closeZombieTask(task));
                 it.remove();
             } catch (final RuntimeException e) {
                 log.error("Suspending {} {} failed due to the following error:", taskTypeName, task.id(), e);
+                exception.compareAndSet(null, e);
```

Comment:
Hmm.. is this correct? If we get an exception `e`, it will be recorded here, and the fatal exception `f` below will never be recorded. I.e. line 230 is doomed to be a no-op right?

Reply:
That's true. We can remove the second `compareAndSet` -- we can only keep the first exception here and just log all others.